fineract-core/src/main/java/org/apache/fineract/portfolio/savings/data/SavingsProductData.java

```diff
@@ -590,19 +590,21 @@ public boolean isWithdrawalFeeForTransfers() {
     }
 
     public boolean isCashBasedAccountingEnabled() {
-        return AccountingRuleType.CASH_BASED.getValue().toString().equals(this.accountingRule.getValue());
+        return this.accountingRule != null && AccountingRuleType.CASH_BASED.getValue().toString().equals(this.accountingRule.getValue());
     }
 
     public boolean isAccrualBasedAccountingEnabled() {
         return isUpfrontAccrualAccounting() || isPeriodicAccrualAccounting();
     }
 
     public boolean isUpfrontAccrualAccounting() {
-        return AccountingRuleType.ACCRUAL_UPFRONT.getValue().toString().equals(this.accountingRule.getValue());
```

Comment:
@hiteshprajapati if you intend to push this back to the community then you need to add atleast a unit test to it.